Sources/App/CRP/SlackCommand+CRP.swift

```diff
@@ -69,3 +68,41 @@ extension SlackCommand {
         })
     }
 }
+
+/// Filter the CHANGELOG entries then orders and format the CHANGELOG text
+///
+/// - Parameters:
+///   - commits: The list of commits gathered between last release and current one
+///   - release: The release for which to build the CHANGELOG text for
+/// - Returns: The text containing the filtered and formatted CHANGELOG, grouped and ordered by jira board
+private func formatChangelog(using commits: [String], for release: GitHubService.Release) -> String {
+    // Only keep SDK commits if release is for SDK
+    let filteredMessages = release.isSDK ? commits.filter { $0.contains("#SDK") || $0.contains("[SDK-") } : commits
+
+    // Group then sort the changes by JIRA boards (unclassified last)
+    let grouped = Dictionary(grouping: filteredMessages) { ticket(from: $0)?.board }
+        .sorted { e1, e2 in e1.key ?? "ZZZ" < e2.key ?? "ZZZ" }
+
+    // Build a CHANGELOG text
+    return grouped
+        .reduce(into: [], { (accum: inout [String], entry: (key: String?, value: [String])) in
+            let title = entry.key.map { "\($0) tickets" } ?? "Other"
+            accum.append("## \(title)")
+            accum.append("")
+            accum.append(contentsOf: entry.value)
+            accum.append("")
+        })
+        .joined(separator: "\n")
+}
+
+private let jiraBoardRegex = try! NSRegularExpression(pattern: #"\b([A-Za-z]*)-[0-9]*\b"#, options: [])
+
+private func ticket(from message: String) -> (board: String, key: String)? {
```

Comment:
what the `key` stands here for?

Reply:
for `[ABC-123]`, `board` is `"ABC"` and `key` is `"ABC-123"`.
 - I've named it "key" because it's the name of the field in the JIRA API response (e.g. see [this API response](https://babylonpartners.atlassian.net/rest/api/3/issue/216051))
 - Admittedly I'm not using it yet in this PR, but I'll very soon need it in the next PR when I'll have to iterate thru all the tickets mentioned in the CRP ticket in order to set their "Fixed Version" field, that's why I've taken the occasion to already extract it

Reply:
maybe it would be better to rename `key` to `ticketNumber` or something?

Reply:
Yeah, why not 😉 

I'm not sure `ticketNumber` is unambiguous either as the "number" part feels like it would only contain `123` not `ABC-123`. Especially since I have the `board` already separated which could lead to think `ticketNumber` is just "the rest" (while I'll never _just_ need `123` in the future features, only the ticket `key` – so that I can then use the REST API with `"https://babylonpartners.atlassian.net/rest/api/3/issue/\(key)"` to update the Fixed Version later).

I'm totally ok for changing the name, just not sure which could be best to describe that precisely then (naming is hard) 🤔

Reply:
What about just `ticket`? or `ticketName` or `ticketKey`? (not `ticketID` since `ID` is a different field in JIRA)

Reply:
ticket "number" is commonly used for that I think, but "id" would be fine. It would make sense to wrap the whole return value of this method in the `Ticket` structure (or typealias), then key can become `id` there

Reply:
Lol that's what I did at first (`typealias Ticket = (board: String. key: String)`) but removed it before pushing because I since it was only used at one single place (the signature of this function) I felt it wasn't worth it. But I can re-add it, I agree it would make things clearer

Reply:
@ilyapuchka Made the change in 2ef707c637c342ed01c8efe2618c00470be75a99 ; ended up creating a simple but dedicated `struct JiraService.TicketID`

Reply:
Yes, makes total sense!